src/models/Orders.js

```diff
@@ -0,0 +1,105 @@
+const mongoose = require('mongoose');
+
+// mongoose schema for the individual order items
+const orderItemSchema = new mongoose.Schema({
+    // The album field references the Album model and is required
+  album: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Album', // Referencing the Album model
+    required: true,
+  },
+  // The quantity field specifies the number of items in the order
+  quantity: {
+    type: Number,
+    required: true, // Quantity is required
+    min: 1, // Minimum quantity allowed is 1
+  },
+});
+
+// Create the OrderItem model using the defined schema
+const OrderItem = mongoose.model('OrderItem', orderItemSchema);
+
+// Schema for the main Order model
+const orderSchema = new mongoose.Schema({
+    // The user field references the User model and is required
+  user: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User', // Referencing the User model
+    required: true,
+  },
+  // The orderStatus field indicates the current status of the order
+  orderStatus: {
+    type: String,
+    enum: ['pending', 'payment_successful', 'payment_failed', 'cancelled', 'complete'],
+    required: true, // Order status is required
+    default: 'pending' // Set initial order status as pending
+  },
+  // The tax field represents the tax percentage applied to the order
+  tax: {
+    type: Number,
+    required: true, // Tax percentage is required
+    min: 0, // Tax percentage must be at least 0
+    max: 1, // Tax percentage must be at most 1
+  },
+  // The subtotal field represents the total price before taxes
+  subtotal: {
+    type: Number,
+    required: true,
+  },
+  // The total field represents the total price after taxes
+  total: {
+    type: Number,
+    required: true,
+  },
+  // The paymentIntentId field stores the Stripe state of the purchase
+  paymentIntentId: String, // Payment intent ID is a string
+}, {
+  timestamps: true, // Automatically add createdAt and updatedAt timestamps
+});
+
+// Create a function to update order statuses
+const updateOrderStatus = async () => {
+  try {
+    // Get the current date and subtract 1 minute
+    const oneMinutesAgo = new Date(Date.now() - 1 * 60 * 1000);
+    
+    // Update orders with a status of "pending" that were created 1 minute ago or earlier,
+    // and set the order status to "cancelled"
+    const result = await Order.updateMany(
+      { orderStatus: 'pending', createdAt: { $lte: oneMinutesAgo } },
+      { $set: { orderStatus: 'cancelled' } }
+    );
```

Comment:
We can leave it like this for now for testing; but that's a pretty short time to cancel an order :sweat\_smile: . I can imagine a situation as a user where  I proceeded to Checkout (order is created) and then had to go and find my credit card, and it took me more than 1 minute and I come back and see that my order has been set to "cancelled".

Reply:
This was only for testing purposes, as I was on the verge of falling asleep every 10 minutes while testing😴. I will fix this once we're done with the shopping cart and payments.